plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -127,7 +127,20 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 	}
 
 	trace.Step("Selecting host")
-	return g.selectHost(priorityList)
+	host, err := g.selectHost(priorityList)
+	if err != nil {
+		return "", err
+	}
+
+	if len(g.extenders) != 0 {
+		for _, extender := range g.extenders {
+			err := extender.Bind(pod, host)
+			if err != nil {
```

Comment:
I haven't run in to >1 use case but here is how extenders recover. On any Bind failure, the pod is not scheduled and scheduler will attempt to schedule the pod again a little bit later. This results in another Bind on the same pod UID. Extenders that have reserved resources in the earlier incarnation can free them up when that happens.